### ISSUE TITLE
emit _classCallCheck

### ISSUE THREAD

**Issue by basarat** [microsoft/TypeScript#6569]
For the follwing: 

``` ts
class Foo{ 
}
```

[Babel emits](https://babeljs.io/repl/#?experimental=false&evaluate=true&loose=false&spec=false&code=class%20Foo%7B%0A%20%20%0A%7D): 

``` js
"use strict";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var Foo = function Foo() {
  _classCallCheck(this, Foo);
};
```

Not really needed for TypeScript. But a request for a similar emit was made for TypeScript : http://stackoverflow.com/q/34945858/390330 

Purely for tracking :rose:

**Comment by riggerthegeek** [comment]
As the author of the SO post, my use-case is for times when writing a library that may or may not be consumed by TypeScript.  Clearly, this should be off by default, but it would be useful to be able to throw a runtime error if the function wasn't called with `new` (as in Babel).

It would also be useful to have this flag throw runtime errors for as many of the TypeScript compiler errors as is practical (checking that an object fulfils an interface is probably impossible, but checking that the input is an object is achievable).

**Comment by mhegazy** [comment]
with https://github.com/Microsoft/TypeScript/issues/5595 in place, such transformations should be easy to do as extensions to the compiler. marking this as revisit for now.

**Comment by RyanCavanaugh** [comment]
Closing due to lack of additional feedback
